src/components/dropdown/CustomRender.tsx

```diff
@@ -35,44 +34,33 @@ export const DefaultOptionRenderer = (props: any) => {
     data: props.data,
     inputValue: props.selectProps?.inputValue,
   };
-
   return isSearchHeaderOption && enableTermSearch ? (
-    inputValue && (
+    inputValue && 
       <components.Option {...props}>
         <HeaderComp {...props} />
       </components.Option>
-    )
-  ) : (
-    <>
-      {OptionRenderer ? (
-        <div className="tk-option">
-          <components.Option {...props}>
-            <OptionRenderer {...rendererProps} />
-          </components.Option>
-        </div>
-      ) : (
-        <div className="tk-option">
-          <components.Option {...props} className={classNames(classNamePrefix && mode ? `${classNamePrefix}__option--${mode}` : null)}/>
-        </div>
-      )}
-    </>
-  );
+  ) : <>
+    {OptionRenderer ? 
+      <div className="tk-option">
+        <components.Option {...props}>
+          <OptionRenderer {...rendererProps} />
+        </components.Option>
+      </div>
+      : <div className="tk-option">
+        <components.Option {...props} className={classNames(classNamePrefix && mode ? `${classNamePrefix}__option--${mode}` : null)}/>
+      </div>
+    }
+  </>;
 };
 
 // Specific Input to fix input not displayed in React-Select
 // See https://github.com/JedWatson/react-select/issues/3068
 // See https://github.com/JedWatson/react-select/discussions/4302
 export const Input = (props: any) => {
   const inputAlwaysDisplayed = props?.selectProps?.inputAlwaysDisplayed;
-  return inputAlwaysDisplayed ? (
-    <components.Input
-      {...props}
-      isHidden={false}
-      onKeyUp={props?.selectProps?.onKeyUp}
-    />
-  ) : (
-    <components.Input {...props} onKeyUp={props?.selectProps?.onKeyUp} />
-  );
+  return inputAlwaysDisplayed ? 
+    <components.Input {...props} isHidden={false} onKeyUp={props?.selectProps?.onKeyUp}/>
```

Comment:
we're repeating this line just to set isHidden=false? 
why not isHidden={!inputIsAlwaysVisible}

Reply:
done

Reply:
`inputAlwaysDisplayed` is used by TimePicker. I'm not sure the isHidden attribute should be given by default. It may impact the default behavior of Dropdown. I would you suggest to test if there is no regression in this case.

Reply:
Actually by default is false.

Reply:
Updated also